bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,388 @@
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YioRemoteDockHandleStatus;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YioRemoteMessages;
+import org.openhab.binding.yioremote.internal.dtos.AuthenticationMessage;
+import org.openhab.binding.yioremote.internal.dtos.IRCode;
+import org.openhab.binding.yioremote.internal.dtos.IRCodeSendMessage;
+import org.openhab.binding.yioremote.internal.dtos.IRReceiverMessage;
+import org.openhab.binding.yioremote.internal.utils.Websocket;
+import org.openhab.binding.yioremote.internal.utils.WebsocketInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link YIOremoteDockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Loercher - Initial contribution
+ */
+@NonNullByDefault
+public class YIOremoteDockHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(YIOremoteDockHandler.class);
+
+    YIOremoteConfiguration localConfig = getConfigAs(YIOremoteConfiguration.class);
+    private WebSocketClient webSocketClient = new WebSocketClient();
+    private Websocket yioremoteDockwebSocketClient = new Websocket();
+    private ClientUpgradeRequest yioremoteDockwebSocketClientrequest = new ClientUpgradeRequest();
+    private @Nullable URI websocketAddress;
+    private YioRemoteDockHandleStatus yioRemoteDockActualStatus = YioRemoteDockHandleStatus.UNINITIALIZED_STATE;
+    private @Nullable Future<?> webSocketPollingJob;
+    public String receivedmessage = "";
+    private JsonObject recievedjson = new JsonObject();
+    private boolean heartbeat = false;
+    private boolean authenticationok = false;
+    private String receivedstatus = "";
+    private String lastsendircode = "";
```

Comment:
Please double check the capitalization, as mentioned in the previous reviews.

Reply:
resolved